src/domains/estoque/reserve/ReservaTecnico/index.js

```diff
@@ -21,87 +24,79 @@ const { Op } = Sequelize
 
 class ReservaTecnicoDomain {
   async create(body, options = {}) {
+    let fieldsError = []
     const { transaction = null } = options
-
-    const reservaTecnicoNotHasProp = prop => R.not(R.has(prop, body))
-
-    const field = { technician: false }
-
-    const message = { technician: '' }
-
-    let errors = false
-
-    let technicianId = ''
-
-    if (reservaTecnicoNotHasProp('data')) {
-      field.data = true
-      message.data = 'data cannot undefined'
-      errors = true
+    const payload = {
+      technician: pathOr(null, ['technician'], body),
+      rows: pathOr(null, ['rows'], body),
+      data: pathOr(null, ['data'], body),
+      accessSecurity: pathOr(null, ['accessSecurity'], body),
     }
-
-    if (reservaTecnicoNotHasProp('technician') || !body.technician) {
-      field.technician = true
-      message.technician = 'technician cannot null'
-      errors = true
-    } else {
-      const tecnico = await Technician.findOne({
-        where: { name: body.technician },
-        transaction
-      })
-
-      if (!tecnico) {
-        field.technician = true
-        message.technician = 'técnico inválido'
-        errors = true
-      } else {
-        technicianId = tecnico.id
+    
+    for ( let key in payload) {
```

Comment:
o valor de key não é alterado então poderia ser declarado como uma const

Reply:
não tenho certeza se funciona já que cada ver que roda o for a key recebe um valor diferente, no primeiro momento ela é um name no  segundo um status e assim por diante mas irei testar